Minio/MinioClient.cs

```diff
@@ -437,6 +441,11 @@ public MinioClient WithRetryPolicy(RetryPolicyHandlingDelegate retryPolicyHandle
     /// <returns></returns>
     public MinioClient WithHttpClient(HttpClient httpClient)
     {
+        if (HTTPClient != null)
```

Comment:
This function should throw an exception for httpClient bring null which is the input as InvalidArgumentException

Reply:
Also this If condition is not needed unless we kind of support all kinds of constructors, we should perhaps remove old style constructors (deprecated ones), move to 5.0.0 and drop all of their support.

Reply:
Done.